modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/utilities/quantities.js

```diff
@@ -25,19 +25,40 @@ export function getMinQuantity(
 
 export function getMultiple(
 	incrementalOrderQuantity= 1,
-	multipleQuantity = 1
+	multipleQuantity = 1,
+	precision = 0
 ) {
-	if (incrementalOrderQuantity === multipleQuantity) {
-		return incrementalOrderQuantity;
+	let precisionAdjustment = 0;
+
+	if (incrementalOrderQuantity < 1 && multipleQuantity < 1) {
+		let num =
+			(incrementalOrderQuantity < multipleQuantity) ?
+				incrementalOrderQuantity : multipleQuantity;
+
+		while (num < 1) {
+			num *= 10;
+			precisionAdjustment++;
+		}
+	}
+
+	let unitOfMeasureMultiple = incrementalOrderQuantity;
+	let itemMultiple = multipleQuantity;
+
+	if (precisionAdjustment > 0) {
+		unitOfMeasureMultiple =
+			incrementalOrderQuantity * 10 * precisionAdjustment;
+		itemMultiple =
+			multipleQuantity * 10 * precisionAdjustment;
 	}
 
-	//this solves for least common multiple
 	let multiple = incrementalOrderQuantity;
-	while (multiple % multipleQuantity !== 0) {
-		multiple += incrementalOrderQuantity;
+
+	while (multiple % itemMultiple !== 0) {
+		multiple += unitOfMeasureMultiple;
```

Comment:
@wanderlast is it not easier to use something like this?
function findLCM(a, b) { 
    let lar = Math.max(a, b); 
    let small = Math.min(a, b); 
    for (i = lar; ; i += lar) { 
        if (i % small == 0) 
            return i; 
    } 
} 
will this not solve for all cases?

Reply:
Like I mentioned, %/modulus when used with two decimal values below 1 causes the portal to infinite loop/lag out due to it technically never reaching a valid value when evaluating i % small. You can see an example of this in the screenshot below where I used the function you provided:

![Screenshot 2024-07-12 115653](https://github.com/user-attachments/assets/951232e3-cad1-40ae-a361-08bb269427a8)

Note the variable values on the left.